File: src/lib/software/discord.py
from src.lib.software import BaseSoftware, SoftwareCategory
from src.lib.software.variants.discord_ptb import DiscordPtb
from src.lib.software.variants.discord_stable import DiscordStable
from src.lib.software.variants.discord_canary import DiscordCanary

class Discord(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'discord'
        self.name = 'Discord'
        self.category = [SoftwareCategory.Social]
        self.variants = [DiscordStable(), DiscordPtb(), DiscordCanary()]
        self.icon = 'discord.png'
        self.homepage = 'https://discord.com'

    def resolve_download_url(self):
        pass

File: packages/app/src/services/software/definitions/discord.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'discord';
	public name = 'Discord';
	public category = [SoftwareCategory.Social];
	public downloadName = '';
	public icon = 'discord.png';
	public homepage = 'https://discord.com';

	public async resolveDownloadUrl() {
		return ok('');
	}
}
